Bottom: f600af05289e413bef6b724a0bc9235a09d6aa47
Top:    511d59a4118b97b791c3953b1998887da29858e2
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-16 14:49:12 +0800

autofs-5.0.7 - only prove specific nfs version if requested


---

diff --git a/modules/mount_nfs.c b/modules/mount_nfs.c
index adf2002..a1e2d40 100644
--- a/modules/mount_nfs.c
+++ b/modules/mount_nfs.c
@@ -146,7 +146,15 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 				if (strncmp("vers=4", cp, o_len) == 0 ||
 				    strncmp("nfsvers=4", cp, o_len) == 0)
 					vers = NFS4_VERS_MASK | TCP_SUPPORTED;
-				else if (strstr(cp, "port=") == cp &&
+				else if (strncmp("vers=3", cp, o_len) == 0 ||
+					 strncmp("nfsvers=3", cp, o_len) == 0) {
+					vers &= ~NFS4_VERS_MASK;
+					vers = (NFS3_REQUESTED & ~NFS_VERS_MASK);
+				} else if (strncmp("vers=2", cp, o_len) == 0 ||
+					 strncmp("nfsvers=2", cp, o_len) == 0) {
+					vers &= ~NFS4_VERS_MASK;
+					vers |= (NFS2_REQUESTED & ~NFS_VERS_MASK);
+				} else if (strstr(cp, "port=") == cp &&
 					 o_len - 5 < 25) {
 					char optport[25];
